Fix constraint sensitivities and LCOE drivers discussion; rewrite QP convexity appendix (#189)
symbiotic-engineering/MDOcean: pubs/applied-ocean-research-model/sections/discussion.tex

From \cref{eq:constrained-qp-solution,eq:power-elec}, the average electrical power (the LCOE denominator) is a sea-state-weighted sum of the impedance-matched power in each sea state, each scaled down by the factor $(1-|\Gamma_{\beta,opt}|^2)$ that captures the effect of active dynamic constraints:
\begin{equation}\label{eq:power-elec-sum}
\begin{aligned}
    &\overline{P}_{elec} = \eta \sum_{\beta=1}^{N_{sea}} JPD_\beta \cdot 
    \frac{|\hat{V}_{s,th,\beta}|^2}{8\Re(Z_{s,th,\beta})}~ (1 - |\Gamma_\beta|^2)
    \overline{P}_{elec} &= \eta \sum_{\beta=1}^{N_{sea}} JPD_\beta \cdot
    \frac{|\hat{V}_{s,th,\beta}|^2}{8\Re(Z_{s,th,\beta})}~ (1 - |\Gamma_{\beta,opt}|^2)
    \\ s.t. \quad
    |\hat{F}_{p,\beta}| &\leq F_{max},\\
    |p_{pk,elec,\beta}| &\leq P_{max,elec},\\
    |\hat{\xi}_{f,\beta}| &\leq X_{f,max},\\
    |\hat{\xi}_{s,\beta}| &\leq X_{s,max}\\
    |\hat{F}_{p,\beta}| &\leq F_{max},\quad
    |p_{pk,elec,\beta}| \leq P_{max,elec},\\
    |\hat{\xi}_{f,\beta}| &\leq X_{f,max},\quad
    |\hat{\xi}_{s,\beta}| \leq X_{s,max}\\
\end{aligned}
\end{equation}

> Aggregating the reflection coefficients for all sea states into a vector $\vec{\Gamma}$, the effect of the constraints on power can be expressed as the difference between the unconstrained power and the power lost from impedance mismatch required to obey constraints:

Reviewer: start deleted without moving

\begin{equation}
\begin{aligned}
    \overline{P}_{elec} = \overline{P}_{elec,unconstrained} (1 - \vec{\Gamma}^* \, \mathbf{I} \, \vec{\Gamma}) \\
    s.t. \quad S_\mu (\vec{\Gamma} - \vec{\Gamma}_{c,\mu})^* \, \mathbf{I} \, (\vec{\Gamma} - \vec{\Gamma}_{c,\mu}) 
    \leq S_\mu \vec{r}_\mu^{~2} 
    \\ \forall \mu \in \{1,...,N_{constraints}\}
\end{aligned}
\end{equation}

> where $\mathbf{I}$ is the identity matrix of size $N_{sea}\times N_{sea}$.

Reviewer: end deleted without moving

> Power is maximized when $|\Gamma_{\beta,opt}|=0$ in every sea state, which corresponds to unconstrained impedance-matched control with no active constraints.

Reviewer: ```suggestion
Power is maximized when $|\Gamma_{\beta,opt}|=0$ in every sea state, which corresponds to unconstrained impedance-matched control.
```

> As derived in \cref{sec:appendix-constraint-sensitivity}, the constraint limits $F_{max}$, $P_{max,elec}$, $X_{f,max}$, and $X_{s,max}$ affect only the radii $r_\mu$ of the constraint circles in the complex $\Gamma$ plane (see \cref{sec:appendix-qp-solution}), not the circle centers $\Gamma_{c,\mu}$.

Reviewer: move to appendix, here it should just say that the form is piecewise affine + sqrt(affine or quadratic) which is (?) convex

Larger constraint limits produce larger circle radii, which allow a smaller optimal $|\Gamma_{\beta,opt}|$ and thus monotonically increase power.
The optimal reflection coefficient in each sea state takes one of the following values depending on which constraints are active (see \cref{sec:appendix-constraint-sensitivity} for the complete derivation):
\begin{equation}\label{eq:gamma-opt-summary}
    |\Gamma_{\beta,opt}| =
    \begin{cases}
        0 & \text{if no constraints active} \\
    |\vec{\Gamma}_{c,\mu^*}| - \vec{r}_{\mu^*} & \text{if constraint $\mu^*$ active, others inactive} \\
     & \text{if constraints $\mu^*$, $\nu^*$ active, others inactive} \\
        0 & \text{no constraints active in sea state $\beta$} \\
        |\Gamma_{c,\mu^*}| - r_{\mu^*} & \text{one constraint $\mu^*$ active} \\
        |\Gamma_{\mu^*,\nu^*}| & \text{constraints $\mu^*$ and $\nu^*$ both active}
    \end{cases}
\end{equation}

> where $*$ denotes the active constraint(s) in each sea state.

Reviewer: start deleted without moving

Power scales with $b_\mu$ and $b_{\nu}$ as follows:
\begin{equation}
    P_{elec} = P_0 + p_\mu b_\mu + p_\nu b_\nu \pm p_{\mu\nu} \sqrt{ a_0 b_\mu^2 + a_1 b_\nu^2 + a_2 b_\mu b_\nu + a_3 b_\mu + a_4 b_\nu + a_5}

> \end{equation}

Reviewer: end  deleted without moving

Reviewer: I got (derived in appendix, eq:power-double-sum-main) quadratic + affine*sqrt(quadratic) + affine, vs this eqn (derived earlier) got affine + sqrt(quadratic). Worth double checking later.

Reviewer: I think my eqn in appendix is correct for scaling of power = 1-|gamma|^2, and old eqn is correct for scaling of just |gamma|.

where $\Gamma_{\mu^*,\nu^*}$ is the intersection point of the two active constraint circles.

> We are interested in the sensitivity of power to the constraint limits $F_{max}$, $P_{pk,elec}$, $X_{f,max}$, and $X_{s,max}$.

Reviewer: start  moved (ok, check diff)

Considering that the active set varies with the constraint limits, the global dependence of power on constraint limit can be found by integrating the local sensitivity across the range of constraint limits, and will be convex if the original problem is convex.
Thus, for cases where two constraints are active and the optimal occurs at an intersection of the constraint circles, the average power is piecewise affine plus square root of quadratic in the constraint limits.
For cases where only one constraint is active, the average power is piecewise affine plus square root of affine in the constraint limits.

Plugging these optimal solutions into the power equation gives the following sensitivitiy, which applies globally as long as $\mu$ and $\nu$ are adjusted to reflect any change in active constraints that occurs as limits are varied:
\begin{equation}
    \overline{P}_{elec} = \overline{P}_{elec,unconstrained} - \sum_{\beta=1}^{N_{sea}} JPD_\beta \cdot 
    \frac{|\hat{V}_{s,th,\beta}|^2}{8\Re(Z_{s,th,\beta})}~ (1 - |\Gamma_\beta|^2)

> \end{equation}

Reviewer: end moved (check diff)

> Since the power maximization problem is convex after reduction to the $\Gamma$ parameterization (see \cref{sec:qp-convexity}), integrating these local sensitivities gives a global power--limit relationship that is also convex.

Reviewer: ```suggestion
Since the unconstrained power maximization problem is convex after reduction to the $\Gamma$ parameterization (see \cref{sec:qp-convexity}), integrating these local sensitivities gives a global relationship between power and constraint limits that is also convex.
```

\begin{equation}\label{eq:power-elec-sum}
\begin{aligned}
    \overline{P}_{elec} &= \eta \sum_{\beta=1}^{N_{sea}} JPD_\beta \cdot
    \frac{|\hat{V}_{s,th,\beta}|^2}{8\Re(Z_{s,th,\beta})}~ (1 - |\Gamma_{\beta,opt}|^2)
    \\ s.t. \quad
    |\hat{F}_{p,\beta}| &\leq F_{max},\quad
    |p_{pk,elec,\beta}| \leq P_{max,elec},\\
    |\hat{\xi}_{f,\beta}| &\leq X_{f,max},\quad
    |\hat{\xi}_{s,\beta}| \leq X_{s,max}\\
\end{aligned}
\end{equation}
Power is maximized when $|\Gamma_{\beta,opt}|=0$ in every sea state, which corresponds to unconstrained impedance-matched control.
When constraints are active, the optimal $|\Gamma_{\beta,opt}|>0$ represents the impedance mismatch required to satisfy them, 
and power decreases quadratically with $|\Gamma_{\beta,opt}|$.
 
\Cref{sec:appendix-constraint-sensitivity} uses this relationship to derive the scaling law between the optimal annual average power $\overline{P}_{elec}$ and the constraint coefficients $b_\mu$:
\begin{equation}\label{eq:power-double-sum-main}
\begin{aligned}
    \overline{P}_{elec} = \overline{P}_{elec,0}
    - \eta \sum_{\mu\nu\beta}
    &\left[

>         \sqrt{\textrm{aff}_{\mu\nu\beta}(b_{\mu})}

Reviewer: eventually show this as square of aff + sqrt(aff) + sqrt(quad), less complicted

        + \textrm{quad}_{\mu\nu\beta}(b_{\mu}, b_{\nu}) 
    \right.+ 
     \\
    &\left.
        \left(\textrm{aff}_{\mu\nu\beta}(b_{\mu}) + \textrm{aff}_{\mu\nu\beta}(b_{\nu})\right)
        \sqrt{\textrm{quad}_{\mu\nu\beta}(b_{\mu}, b_{\nu})}
    \right]
\end{aligned}\end{equation}
where $b_\mu$ and $b_\nu$ are simple (affine or square) functions of the constraint limits

> $F_{max}$, $P_{max,elec}$, $X_{f,max}$, and $X_{s,max}$ defined in \cref{tab:qp-constraints}; 

Reviewer: ```suggestion
$\tau_{max}$, $P_{\tau\Omega,max}$, $\xi_{f,max}$, $\xi_{s,max}$, and $X_{PTO,max}$ defined in \cref{tab:qp-constraints}; 
```

More strongly, the presence of the PTO limits in the numerator and denominator but not elsewhere in the model implies that the system optimal will have both PTO limits active in at least one sea state.
The numerator is affine in the PTO limits whlie the denominator is piecewise affine plus square root of quadratic or affine in the PTO limits.
The numerator is affine in the PTO limits while the denominator is piecewise affine plus square root of quadratic or affine in the PTO limits.
The LCOE minimization with design variables of $F_{max}$ and $P_{max,elec}$ and thicknesses is considered a fractional quadratic program, a structure common in resource allocation problems.
The convexity properties of this problem can motivate a double-nested strategy for optimization: at the innermost level, the optimal control problem for fixed geometry and PTO limits can be solved as an analytical QP, then the mid-level optimization can be solved as a low-dimensional fractional quadratic program to find the optimal PTO limits, and the outer-level optimization can be solved as a general nonlinear program to find the optimal geometry.

> LCOE with gamma:

Reviewer: start: deleted without moving

\begin{equation}
    LCOE \sim \frac{p'_0+p'_1x_1 + p'_2x_2 + p'_3x_3}{\bar{P}_{elec,unconstrained} - \vec{|\Gamma|} \mathbf{Q}_\Gamma \vec{|\Gamma|}}
\end{equation}

LCOE with x for x=Fmax, Ppk, thickness:
- $p'_n$ are constant
- $\mathbf{Q}_\Gamma$ is constant
- $\bar{P}_{elec,unconstrained}$ is constant
- $|\Gamma|$ is constant if the amplitude constraint is active, and the following if an effort limit is active:
\begin{equation}
    |\Gamma| \sim |\Gamma_{c,\mu}| - r_{\mu} \\
\end{equation}
plug in $|\Gamma_{c,\mu}|$ and $r_{\mu}$:
\begin{equation}
    |\Gamma| = \frac{a_1 e_{max}^2 + a_0 \pm 
    \sqrt{(a_1 e_{max}^2+a_0)^2 + (a_3 e_{max}^2 + a_4)(a_5 e_{max}^2 + a_6)}}
    {a_3 e_{max}^2 + a_4}
\end{equation}
a function of the design variables, with the following dependence on the design variables:

LCOE with x for x=Fmax, Ppk, thickness, and bulk dimensions:
- $p'_n$ are constant for n=0, 1, 2
\begin{equation}
    p'_3 \sim x_4^2, \qquad
\end{equation}

> \hl{write Fmax and Ppk and Pavg in terms of the control variables, showing the lcoe tradeoff of the optimal control problem / pto sizing problem for a given mechanical design.}

Reviewer: end deleted without moving

> The convexity of the denominator with respect to the PTO limits (established in the preceding paragraph) means that LCOE is quasi-convex in the PTO limits for fixed geometry, guaranteeing a unique global optimum in those variables.

Reviewer: reword

Reviewer: I'm not sure that quasi convex is accurate

Reviewer: https://web.stanford.edu/~boyd/papers/pdf/dqcp.pdf says "Hence the ratio of a nonnegative convex function and a positive concave function is quasiconvex, and the ratio of a nonnegative concave function and a positive convex function is quasiconcave"

> This motivates a nested optimization strategy: at the innermost level the optimal control problem for fixed geometry and PTO limits is solved as an analytical QP; at the mid-level the optimal PTO limits are found by exploiting the quasi-convex fractional structure; and at the outer level the bulk geometry is optimized with a general nonlinear program.

Reviewer: ```suggestion
This motivates a three-level nested optimization strategy: at the innermost level the optimal control problem for fixed geometry and PTO limits is solved as an analytical QP; at the mid-level the optimal PTO limits are found by exploiting the quasi-convex fractional structure; and at the outer level the bulk geometry is optimized with a general nonlinear program.
```

    }{\overline{P}_{elec}}
\end{equation}

The numerator of \cref{eq:LCOE-scale} is affine in the design inputs $F_{max}$ and $P_{max,elec}$, 
while $V_{struct}$ is the product of structural thicknesses with their corresponding areas, 
which depend quadratically on the bulk dimensions.
The fact that decreasing a constraint limit causes the numerator of the LCOE to monotonically increase 
and the denominator to monotonically decrease implies that at system optimality, 
and each constraint will be active in at least one sea state.
The LCOE minimization with design variables of $F_{max}$, $P_{max,elec}$, and thicknesses is considered a fractional quadratic program, a structure common in resource allocation problems.
When the denominator is convex with respect to the PTO limits, this means that LCOE is ``quasi-convex'' in the PTO limits for fixed geometry \cite{agrawal_disciplined_2020}.

> This quasi-convexity can be leveraged by using a subgradient method or by solving a simple sequence of convex problems, which simplifies the optimization process significantly compared to a general nonlinear solver.

Reviewer: ```suggestion
This quasi-convexity can be leveraged by using a subgradient method or by solving a small sequence of convex problems, which simplifies the optimization process significantly compared to a general nonlinear solver.
```
